Migrate reportback data

angaither opened the pull request:
#### What's this PR do?
- added a function to transform what a phoenix status is to a rogue status
- added script to migrate reportback data to phoenix

#### How should this be reviewed?
👀 

#### Any background context you want to provide?
I still have a few todos/checks
- Once in rogue, does this retry sending back to phoenix? Can we disable that?
- If the script fails and we need to re-migrate all info, the call to `dosomething_rogue_store_rogue_references` will fail because it doesn't check for duplicate rows

#### Relevant tickets
Refs #7225

#### Checklist
- [ ] Documentation added for new features/changed endpoints.
- [ ] Tested on staging.
- [ ] Pinged a PM if this is a larger PR that would benefit from some additional testing love
- [ ] Post a message in #phoenix if this includes something that causes a rebuild!

angaither committed:
Added a skeleton script with a plan for migration.

angaither committed:
Update the data array to include most things.
For now just print it to ensure it's correct.

angaither committed:
Added function to transform phoenix status to rogue status.
Updated the migration query to an inner join. etc.

angaither committed:
Updated script to post data to rogue.

sbsmith86 commented on scripts/export-reportbacks-to-rogue.php:
```diff
@@ -0,0 +1,61 @@
+            INNER JOIN dosomething_reportback rb on rbf.rbid = rb.rbid
+            WHERE rbf.fid > $last_saved
+            ORDER BY rbf.fid
+            LIMIT 1");
```
Does this mean you only get 1 record back?

angaither replied:
oh yeah look at that, I left in the test code

katiecrane commented on scripts/export-reportbacks-to-rogue.php:
```diff
@@ -0,0 +1,61 @@
+  echo 'Migrated reportback ' . $rb->fid . ' to Rogue [' . $rb->caption . ']' . PHP_EOL;
+
+  // Store the reference in dosomething_rogue_reportbacks.
+  dosomething_rogue_store_rogue_references($rb->rbid, $rb->fid, $response);
```
if the post fails I think this is going to break because `$response` won't be anything

angaither replied:
@katiecrane is it nothing, or an is there an error code I should check for?

angaither committed:
Remove limits.
Only save to table if there's a response.

katiecrane commented on scripts/export-reportbacks-to-rogue.php:
```diff
@@ -0,0 +1,62 @@
+
+  $response = $client->postReportback($data);
+
+  if ($response) {
```
@angaither I think I was actually a little wrong before. My understanding is `$response` is only `null` for a `404` and other errors have to be caught otherwise it will crash if it gets an error back when posting. [This is how we do it](https://github.com/DoSomething/phoenix/blob/dev/lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module#L93-L112) for the web form

katiecrane replied:
so it wouldn't crash because it's trying to access stuff that isn't there, but because an error was thrown

sbsmith86 reviewed (approved):
looks good to me!

angaither committed:
Handle multiple times the script runs on phoenix side.
wrap the call to post data in a try/catch

angaither committed:
Update the query to exclude things that already have a reference in rogue.

angaither committed:
Updated the script to
- check for files before sending them to rogue
- only send request to rogue if NS id exists

sbsmith86 commented:
🎉

katiecrane commented on scripts/export-reportbacks-to-rogue.php:
```diff
@@ -0,0 +1,72 @@
+
+      // If the script fails, we can use this to start the script from the last reportback migrated.
+      variable_set('dosomething_rogue_last_rbi_migrated', $rb->fid);
+    } catch (GuzzleHttp\Exception\ServerException $e) {
```
I think you need to catch this kind too `DoSomething\Gateway\Exceptions\ApiException`
